### PR TITLE
Add list method to API docs

### LINKED ISSUE #482: Workers KV API section missing list method
The [Workers API section](https://developers.cloudflare.com/workers/reference/storage/api/#worker-api) is missing the [`list()` method](https://developers.cloudflare.com/workers/reference/storage/listing-keys/).

A short TL;DR with a "see more" link to the main page would help, since a complete list of the in-worker methods would be in one place.

I'm happy to write up a short PR if there's interest :)

### PR DESCRIPTION
**Pull request by GregBrimble**
Fixes #482.

**Changelog**

1. ~Converted existing methods to use standard documentation syntax and highlighting.
    I intentionally left off the expiration options from `put` since they were omitted previously and are already linked out to the page with more info.~
1. Added the `list` method documentation and link to the dedicated page with more info.

### CODE REVIEW

File: content/reference/storage/api.md [hunk 1]
```diff
@@ -227,22 +229,44 @@ Type can be any of:
...
+{{</highlight>}}
 
 The type is automatically inferred from value, and can be any of:
 
-- string
-- ReadableStream
-- ArrayBuffer
+- `string`
+- `ReadableStream`
+- `ArrayBuffer`
```
Comment (GregBrimble) [inline content/reference/storage/api.md]: One thing I queried was the lack of `FormData` type. It shows up in [the actual TypeScript type declarations](https://github.com/cloudflare/workers-types/blob/master/index.d.ts#L168), but no where else that I could see.

**Comment by victoriabernard92** [comment]
As much as I love this PR, I think we want a consistent format for all the API docs and right now this would be the only page using the typescript syntax

I will discuss this PR will my team. If you'd like to get the list-keys in the meantime maybe you could make a separate PR without the typescript?

**Comment by GregBrimble** [comment]
@victoriabernard92, thanks for your feedback!

I've removed the changes to the existing documentation, tried to make the format of the **Listing Keys** section similar, and used the example response object from the `listing-keys.md` page.

Let me know how it's looking, and if there is anything else I can tweak 🙂

File: content/reference/storage/api.md [hunk 2]
```diff
@@ -243,6 +243,22 @@ If you want the keys you write to be automatically deleted at some time in the f
...
+
+```
+{
+  keys: [{ name: "foo", expiration: 1234}],
+  list_complete: false,
+  cursor: "6Ck1la0VxJ0djhidm1MdX2FyD"
+}
+```
+
+For more information about listing keys, and pagination using `cursor` and `list_complete`, see the page [listing keys]({{<ref "listing-keys.md" >}}).
```
Comment (victoriabernard92) [inline content/reference/storage/api.md]:
```suggestion
For more information about listing keys, and pagination using `cursor` and `list_complete`, see the page [listing keys](/reference/storage/listing-keys/).
```
We use absolute refs for future referencing links

**Comment by victoriabernard92** [comment]
@GregBrimble Thank you so much for this! Add comment and then it's ready.

I talked to my team, and if you are interested in converting all the Runtime API docs to TS syntax that would be a huge value add. We just want some consistency so if we do it for one API, we should do it for all.

**Comment by GregBrimble** [comment]
@victoriabernard92 - Changed that link (and two others on the page) to use absolute refs. All good to merge?

**Pull request merged**